### ISSUE TITLE
Switch doesn't work: no rules expected the token `i_`

### ISSUE THREAD

**Issue by Boscop** [Geal/nom#719]
Why am I getting this error?

![image](https://user-images.githubusercontent.com/535593/37553414-f0270012-29c7-11e8-893c-2ece1aacfd49.png)

When I pull the `switch!()` out into a new parser function, it complains about `i` instead of `i_`

![image](https://user-images.githubusercontent.com/535593/37553417-facb3600-29c7-11e8-8c8c-7abac5866eed.png)

The `i_` seems to be coming from here:
https://docs.rs/nom/3.2.1/src/nom/branch.rs.html#423

**Comment by Boscop** [comment]
Ah nevermind, I didn't see the page was longer, I now realized I have to use `call!()`..
It works now :)

Btw the doc is slightly wrong. It says:

> Due to limitations in Rust macros, it is not possible to have simple functions on the right hand side of pattern, like this:
```rust
named!(sw,
   switch!(take!(4),
     b"abcd" => tag!("XYZ") |
     b"efgh" => tag!("123")
   )
 );
...
```
Shouldn't that be this?:
> Due to limitations in Rust macros, it is not possible to have simple functions on the right hand side of pattern, like this:
```rust
named!(sw,
   switch!(take!(4),
     b"abcd" => xyz |
     b"efgh" => num
   )
 );
...
```

**Comment by Geal** [comment]
alright, I fixed the docs in 2943ae5
